pghoard/wal.py

```diff
@@ -42,107 +44,156 @@ class WalBlobLengthError(ValueError):
     """WAL blob is shorter than the WAL header"""
 
 
-WalHeader = namedtuple("WalHeader", ("version", "timeline", "lsn", "filename"))
+WalHeader = namedtuple("WalHeader", ("version", "lsn"))
 
 
-def read_header(blob):
-    if len(blob) < WAL_HEADER_LEN:
-        raise WalBlobLengthError(
-            "Need at least {} bytes of input to read WAL header, got {}".format(WAL_HEADER_LEN, len(blob))
+class LSN:
+
+    SEGSIZE = WAL_SEG_SIZE
+
+    def __init__(self, value, server_version: int, tli: int = None):
```

Comment:
Can we use typing information here (and also for all the return types)?

Reply:
Will do !